server/src/test/java/org/opensearch/action/admin/cluster/node/stats/NodeStatsTests.java

```diff
@@ -1065,4 +1083,257 @@ private static RemoteTranslogTransferTracker.Stats getRandomRemoteTranslogTransf
     private OperationStats getPipelineStats(List<IngestStats.PipelineStat> pipelineStats, String id) {
         return pipelineStats.stream().filter(p1 -> p1.getPipelineId().equals(id)).findFirst().map(p2 -> p2.getStats()).orElse(null);
     }
+
+    public static class MockNodeIndicesStats extends NodeIndicesStats {
```

Comment:
Do we need Mock class ? Can we instead directly use NodeIndicesStats ?

Reply:
I wanted to extract out `statsByIndex`  and `statsByShard` for my testing which is currently not provided as a public method in `NodeIndicesStats`. Instead of adding this to public in the main class, I made these variables protected and extracted it out by extending it.